lib/Codegen/Lowering/StatementLowerer.cpp

```diff
@@ -11,14 +11,15 @@ namespace marco::codegen::lowering
   {
   }
 
-  void StatementLowerer::lower(const ast::Statement& statement)
+  __attribute__((warn_unused_result)) bool StatementLowerer::lower(const ast::Statement& statement)
   {
     if (auto casted = statement.dyn_cast<ast::AssignmentStatement>()) {
       return lower(*casted);
     }
 
     if (auto casted = statement.dyn_cast<ast::BreakStatement>()) {
-      return lower(*casted);
+      lower(*casted);
```

Comment:
In line 21, lower(*casted) returns void, since *casted is a BreakStatement. The outer function returns a bool, which is true if there was no error, and false otherwise. Since lower(*casted) returns void, we cannot simply return its result (since there is no result). Since no error can occur, true is returned.

Reply:
The previous return statement was used to future-proof the code with respect to API changes like the ones you are making. The lower method for BreakStatement should indeed be changed to have a status result, making it coherent with all the others.